pkg/osquery/runtime/osqueryinstance.go

```diff
@@ -537,6 +537,22 @@ func (opts *osqueryOptions) createOsquerydCommand(osquerydBinary string, paths *
 	return cmd, nil
 }
 
+// StartOsqueryClient will create and return a new osquery client with a connection
+// over the socket at the provided path. It will retry for up to 10 seconds to create
+// the connection in the event of a failure.
+func (o *OsqueryInstance) StartOsqueryClient(paths *osqueryFilePaths) (*osquery.ExtensionManagerClient, error) {
```

Comment:
This feels a little awkward. Why wrap NewClient, vs adding some retry logic to newclient directly?

I can roll with it if pressed, but the two different timeouts in play makes this feel weird.

Reply:
I did it this way a) for consistency with [how we start the server](https://github.com/kolide/launcher/blob/main/pkg/osquery/runtime/osqueryinstance.go#L542) and b) because `NewClient` is defined in a [separate package outside this repo](https://github.com/osquery/osquery-go/blob/master/client.go#L35) and it didn't feel worth it to update that for a relatively small change.

Reply:
Oh your right. That does make it messier. Okay